### PR TITLE
Move design system to `web-common`

### PR DESCRIPTION
**Pull request by ericpgreen2**
This PR moves self-contained design system components from `web-local/lib/components` into `web-common/components`. This will make it easier for us to differentiate between lower-level design system components and higher-level application components.

To ensure a clean split, here and going forward, no `web-common` code should import `web-local` code.

### CODE REVIEW

File: web-common/.eslintrc.cjs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module.exports = {
```
Comment (hamilton) [inline web-common/.eslintrc.cjs:1]: we should just have one of these and reference it in each independent directory

**Review by hamilton — COMMENTED** [review]
good start. Seems like there's a bunch of other easy to move / unconnected component sets that should move over in this PR. Is there a reason for the limited set you've moved over vs including the other easy ones?

**Comment by ericpgreen2** [comment]
> good start. Seems like there's a bunch of other easy to move / unconnected component sets that should move over in this PR. Is there a reason for the limited set you've moved over vs including the other easy ones?

Which ones do you have in mind? I could have missed some. I was shooting for components that:
- Don't import anything from `runtime-client` or `web-local`
- Aren't too feature-specific. If it's only used in one place, then I'd lean toward co-locating it with the feature.
- I wouldn't have to refactor. Those requiring refactors I thought I'd handle in follow-on PRs.

Here are my findings from going through what's left in `web-local`:
- `data-graphic` could be moved now. It imports `notifications`, which I originally had in `web-local` but have since moved to `web-common`.
- `data-types` could be moved now.
- `table-editable` is a contender after a small refactor. It currently imports Entity and Validation types from `web-local`, which makes it a bit application-specific. Anyways, for now, it perhaps belong in the "metrics-definition" feature, given it's only used there.
- `virtualized-table` looks good to move over after a small refactor. It has a dependency on `web-local/lib/application-config.ts`'s `TOOLTIP_STRING_LIMIT`.
- `viz` also looks good to move over after a small refactor. It has a dependency on `web-local/lib/application-config.ts`'s `COLUMN_PROFILE_CONFIG`.

File: web-common/.eslintrc.cjs (hunk 1 above)
Comment (ericpgreen2) [reply web-common/.eslintrc.cjs]: Good call, just added a commit.

**Comment by hamilton** [comment]
That set looks good. I think the preview table can be moved over, and we can move the `ConnectedPreviewTable` component to stay in `web-local` since it relies on `runtime-client`.

**Comment by ericpgreen2** [comment]
> That set looks good. I think the preview table can be moved over, and we can move the `ConnectedPreviewTable` component to stay in `web-local` since it relies on `runtime-client`.

Sweet. I like that idea for `preview-table`. Since it's dependent on `virtualized-table`, I'd opt to do it after a follow-on PR that addresses `virtualized-table`'s `web-local` dependency.

**Review by hamilton — APPROVED** [review]

**Pull request merged**